pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,447 @@
+	}
+	namespace = kvs.Items[0].Namespace
+	name = kvs.Items[0].Name
+	return
+}
+
+func hasVerbosityInKV(kv *v1.KubeVirt) (map[string]uint, error) {
+	verbosityMap := map[string]uint{} // key: component name, value: verbosity
+	// check the logVerbosity field in the KubeVirt CR
+	lvJSON, err := json.Marshal(kv.Spec.Configuration.DeveloperConfiguration.LogVerbosity)
+	if err != nil {
+		return nil, err
+	}
+	// if there is a logVerbosity field, input the JSON data to the verbosityMap
+	if err := json.Unmarshal(lvJSON, &verbosityMap); err != nil {
+		return nil, err
+	}
+	return verbosityMap, nil
+}
+
+func createOutputMessage(verbosityVal map[string]uint, options map[string]uint) []string {
+	var messages []string
+	for component := virtAPI; component < all; component++ { // all is the last component, and do not need to check it
+		componentName := getComponentNameByVirtComponent(component)
+		JSONName := getJSONNameByVirtComponent(component)
+		if _, exist := options["all"]; exist {
+			messages = append(messages, fmt.Sprintf("%s=%d", componentName, verbosityVal[JSONName]))
+		} else if _, exist := options[componentName]; exist {
+			messages = append(messages, fmt.Sprintf("%s=%d", componentName, verbosityVal[JSONName]))
+		}
+	}
+	return messages
+}
+
+func createShowMessage(kv *v1.KubeVirt, options map[string]uint) ([]string, error) {
+	// set default verbosity first
+	// it is used to fill the unattended verbosity with default verbosity
+	// key: JSONName, value: verbosity
+	var verbosityVal = map[string]uint{
+		"virtAPI":        virtconfig.DefaultVirtAPILogVerbosity,
+		"virtController": virtconfig.DefaultVirtControllerLogVerbosity,
+		"virtHandler":    virtconfig.DefaultVirtHandlerLogVerbosity,
+		"virtLauncher":   virtconfig.DefaultVirtLauncherLogVerbosity,
+		"virtOperator":   virtconfig.DefaultVirtOperatorLogVerbosity,
+	}
+
+	// if verbosity has been set in the KubeVirt CR, use the verbosity
+	lvMap, err := hasVerbosityInKV(kv)
+	if err != nil {
+		return nil, err
+	}
+	if len(lvMap) > 0 {
+		for key, value := range lvMap {
+			verbosityVal[key] = value
+		}
+	}
+
+	// create a message to show verbosity for the specified component
+	messages := createOutputMessage(verbosityVal, options)
+
+	return messages, nil
+}
+
+func setVerbosity(lvMap map[string]uint, options map[string]uint, patchData *[]patch.PatchOperation, op *string, path *string) {
+	// update lvMap based on the user-specified verbosity for all components
+	if verbosity, exist := options["all"]; exist {
+		for component := virtAPI; component < all; component++ {
+			JSONName := getJSONNameByVirtComponent(component)
+			lvMap[JSONName] = verbosity
+		}
+	}
+	// update lvMap based on the user-specified verbosity for each component
+	for componentName, verbosity := range options {
+		if componentName == "all" {
+			continue
+		}
+		JSONName := getJSONNameByComponentName(componentName)
+		lvMap[JSONName] = verbosity
+	}
+
+	if len(lvMap) != 0 {
```

Comment:
Should probably return an error if `lvMap` ends up being empty.

Reply:
I think we cannot return an error in this case, if `lvMap` ends up being empty, because in the case of just `reset` (no `set` operation after the `reset`) and `currentLv = {}` (no existing verbosity in the KubeVirt CR), this is a valid case of empty `lvMap`. But, we do not need to add another patch in this case. So, I think we cannot return an error.

Although we cannot return an error, we can remove the check `if len(lvMap) != 0 {`, if we do not mind sending duplicate patches in the case of `reset` and `currentLv = {}`.

If we do not allow the reset and set to be set at the same time, the code will be simpler, but user-friendliness will be reduced.